internal/delivery/http/router/router.go

```diff
@@ -0,0 +1,94 @@
+package router
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/go-chi/chi/v5/middleware"
+
+	"github.com/andrsj/feedback-service/internal/delivery/http/middlewares"
+	"github.com/andrsj/feedback-service/internal/infrastructure/cache"
+	"github.com/andrsj/feedback-service/pkg/logger"
+)
+
+type Router struct {
+	router *chi.Mux
+	logger logger.Logger
+
+	cacheMiddleware func(next http.Handler) http.Handler
+	jwtMiddleware   func(next http.Handler) http.Handler
+}
+
+func New(cache cache.Cache, logger logger.Logger) *Router {
+	router := chi.NewRouter()
+
+	router.Use(middleware.Logger)
+	router.Use(middleware.Recoverer)
+
+	jwtMiddleware := middlewares.JWTMiddleware(logger)
+	cacheMiddleware := middlewares.CacheMiddleware(cache)
+
+	return &Router{
+		router:          router,
+		logger:          logger.Named("router"),
+		cacheMiddleware: cacheMiddleware,
+		jwtMiddleware:   jwtMiddleware,
+	}
+}
+
+type Handlers interface {
```

Comment:
This interface looks a bit redundant and could be replaced by the concrete type.